util-collection/src/main/scala/sbt/internal/util/INode.scala

```diff
@@ -35,6 +36,7 @@ abstract class EvaluateSettings[ScopeType]:
   private def getStatic[A](key: ScopedKey[A]): INode[A] =
     static.get(key).getOrElse { sys.error("Illegal reference to key " + key) }
 
+  @nowarn
```

Comment:
Needed to silence

```
[error] -- [E030] Match case Unreachable Error: /home/runner/work/sbt/sbt/util-collection/src/main/scala/sbt/internal/util/INode.scala:47:13 
[error] 47 |        case v: ValidationCapture[a] => strictConstant(v.key: A)
[error]    |             ^^^^^^^^^^^^^^^^^^^^^^^
[error]    |             Unreachable case
[error] -- [E030] Match case Unreachable Error: /home/runner/work/sbt/sbt/util-collection/src/main/scala/sbt/internal/util/INode.scala:48:13 
[error] 48 |        case t: TransformCapture     => strictConstant(t.f: A)
[error]    |             ^^^^^^^^^^^^^^^^^^^
[error]    |             Unreachable case
[error] -- [E030] Match case Unreachable Error: /home/runner/work/sbt/sbt/util-collection/src/main/scala/sbt/internal/util/INode.scala:49:13 
[error] 49 |        case o: Optional[s, A] =>
[error]    |             ^^^^^^^^^^^^^^^^^
[error]    |             Unreachable case
[error] -- [E030] Match case Unreachable Error: /home/runner/work/sbt/sbt/util-collection/src/main/scala/sbt/internal/util/INode.scala:53:13 
[error] 53 |        case StaticScopes => strictConstant(allScopes)
[error]    |             ^^^^^^^^^^^^
[error]    |             Unreachable case
[error] four errors found
```

I tried removing the unreachable case but then `MatchError` is thrown in runtime. So the compiler warning is false.

```
scala.MatchError: sbt.internal.util.Init$ValidationCapture@1ef31f71 (of class sbt.internal.util.Init$ValidationCapture)
	at sbt.internal.util.EvaluateSettings$$anon$2.apply(INode.scala:51)
	at sbt.internal.util.TupleMapExtension$.transform(TupleMapExtension.scala:30)
	at sbt.internal.util.EvaluateSettings$$anon$2.apply(INode.scala:44)
```

c.c. https://github.com/sbt/sbt/actions/runs/11450302490/job/31857503032